pyro/infer/autoguide/guides.py

```diff
@@ -145,6 +145,12 @@ def _update_master(self, master_ref):
         for submodule in self:
             submodule._update_master(master_ref)
 
+    def __setattr__(self, name, value):
```

Comment:
Should this method be moved to the base class `AutoGuide`?

Reply:
Done.